coral/src/app/features/topics/overview/messages/components/TopicMessageItem.tsx

```diff
@@ -0,0 +1,44 @@
+import { BorderBox, Box, Button, Typography } from "@aivenio/aquarium";
+import expandIcon from "@aivenio/aquarium/icons/chevronRight";
+import minimizeIcon from "@aivenio/aquarium/icons/chevronDown";
+import { useState, useId } from "react";
+import truncate from "lodash/truncate";
+
+type Props = {
+  offsetId: string;
+  message: string;
+};
+
+function TopicMessageItem({ offsetId, message }: Props) {
+  const [expanded, setExpanded] = useState<boolean>(false);
+  const panelTrigger = useId();
```

Comment:
TIL about `useId` :o What does it achieve in this context? Asking candidly, I have never seen it used before ^^

Reply:
It's mainly useful when you need to come up with unique id's to link elements accessibility wise. It was included in the React api some versions ago.

Reply:
Also used to some extent in design-system, some examples there as well i think.

Reply:
In general, this component tries to follow the ["accordion" rules](https://www.w3.org/WAI/ARIA/apg/patterns/accordion/examples/accordion/), as this mostly related to it. I almost got a way with just native `details` + `summary` element, but it did not 100% bent to the design.

Reply:
Nice thank you ^^